### PR TITLE
Test Elasticsearch 2.x and 5.x

### LINKED ISSUE #33: Document that the elastic2-doc-manager works with Elasticsearch 5.X
Information posted here https://github.com/mongodb-labs/mongo-connector/issues/453#issuecomment-256103038 should be added to the README and  the MongoConnector Wiki.

### PR DESCRIPTION
**Pull request by ShaneHarvey**
Add tests for MongoDB versions 2.4, 2.6, 3.0, and 3.2 instead of only 2.4. MongoDB 3.4 cannot be tested until mongo-connector 2.5.0. 
Add Elasticsearch 5.x install instructions to the README.

Resolves #33 and #34.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,24 +1,52 @@
...
+  global:
+    - MO_ADDRESS=127.0.0.1:20000
+    - ELASTIC_2_URL=http://download.elastic.co/elasticsearch/release/org/elasticsearch/distribution/tar/elasticsearch/2.4.3/elasticsearch-2.4.3.tar.gz
+    - ELASTIC_5_URL=http://artifacts.elastic.co/downloads/elasticsearch/elasticsearch-5.1.1.tar.gz
+    # We need to manually set JAVA_HOME to Java 8 too.
+    - JAVA_HOME=/usr/lib/jvm/java-8-oracle
+
+matrix:
+  fast_finish: true
+  include:
```
Comment (behackett) [inline .travis.yml]: What about python 3.4?

File: README.rst [hunk 2]
```diff
@@ -1,31 +1,69 @@
...
 `pip <https://pypi.python.org/pypi/pip>`__::
 
   pip install elastic2-doc-manager
 
+Elasticsearch 5.x
+-----------------
+
+For use with an Elasticsearch 5.x server, install with::
+
+  pip install elastic2-doc-manager>=0.3.0 --upgrade 'elasticsearch>=5.0.0,<6.0.0'
```
Comment (behackett) [inline README.rst]: Can we use setuptools extras_require for this?

pip install elastic2-doc-manager[elastic2]

or

pip install elastic2-doc-manager[elastic5]

Comment (ShaneHarvey) [inline README.rst]: We can but then "pip install elastic2-doc-manager" would install without the Elasticsearch dependency. Maybe that's better and in the doc manager we can do:
```python
try:
    import elasticsearch
except ImportError:
    print("Must install with elasticsearch search version 2.x or 5.x:")
    print("pip install elastic2-doc-manager[elastic2]")
    print("pip install elastic2-doc-manager[elastic5]")
```

Comment (ShaneHarvey) [inline README.rst]: Ideally, I'd like to hack setup.py to default to "[elastic2]" as an extra. That way:
pip install elastic2-doc-manager == pip install elastic2-doc-manager[elastic2]
and
pip install elastic2-doc-manager[aws] == pip install elastic2-doc-manager[aws,elastic2]

File: .travis.yml (hunk 1 above)
Comment (ShaneHarvey) [inline .travis.yml]: Oops, added it back.

**Comment by ShaneHarvey** [comment]
Added the Elasticsearch version to `extras_require` so proper installation now requires:
`pip install elastic2-doc-manager[elastic2]`
or
`pip install elastic2-doc-manager[elastic5]`

If a user installs without specifying the version they will get an `ImportError`:
```
>>> from mongo_connector.doc_managers.elastic2_doc_manager import DocManager
Traceback (most recent call last):
  File "/Users/shane/git/elastic2-doc-manager/mongo_connector/doc_managers/elastic2_doc_manager.py", line 29, in <module>
    import elasticsearch
ImportError: No module named 'elasticsearch'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/shane/git/elastic2-doc-manager/mongo_connector/doc_managers/elastic2_doc_manager.py", line 32, in <module>
    "Error: elasticsearch (https://pypi.python.org/pypi/elasticsearch) "
ImportError: Error: elasticsearch (https://pypi.python.org/pypi/elasticsearch) version 2.x or 5.x is not installed.
Install with:
  pip install elastic-doc-manager[elastic2]
or:
  pip install elastic-doc-manager[elastic5]
```

**Pull request merged**